src/modules/commander/Safety.hpp

```diff
@@ -0,0 +1,64 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2022 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file Safety.hpp
+ */
+
+#pragma once
+
+#include <uORB/SubscriptionMultiArray.hpp>
+#include <uORB/Publication.hpp>
+#include <uORB/topics/button_event.h>
+#include <uORB/topics/safety.h>
+
+class Safety
+{
+
+public:
+
+	Safety();
+	virtual ~Safety() = default;
+
+	virtual void safetyButtonHandler();
+	virtual void enableSafety();
+
+private:
+
+	uORB::SubscriptionMultiArray<button_event_s> _safety_button_subs{ORB_ID::safety_button};
```

Comment:
Make sure to limit the number of instances to 2. In this case you could also publish to the same instance with some queue size.
```suggestion
	uORB::SubscriptionMultiArray<button_event_s, 2> _safety_button_subs{ORB_ID::safety_button};
```

Reply:
Thanks for the advice, reworked it to a single instance with the queue. My initial idea was with multi but no need for that.